File: planned/scala/src/main/scala/com/authdog/exceptions/AuthdogException.scala
package com.authdog.exceptions

/**
 * Base exception class for all Authdog SDK errors
 */
class AuthdogException(message: String, cause: Option[Throwable] = None) 
  extends RuntimeException(message, cause.orNull)

object AuthdogException {
  def apply(message: String): AuthdogException = new AuthdogException(message)
  def apply(message: String, cause: Throwable): AuthdogException = new AuthdogException(message, Some(cause))
}

File: planned/php/src/Exceptions/AuthdogException.php
<?php

namespace Authdog\Exceptions;

/**
 * Base exception class for all Authdog SDK errors
 */
class AuthdogException extends \Exception
{
    public function __construct(string $message = "", int $code = 0, ?\Throwable $previous = null)
    {
        parent::__construct($message, $code, $previous);
    }
}
